src/test/librbd/test_Groups.cc

```diff
@@ -480,3 +480,241 @@ TEST_F(TestGroup, add_snapshotPP)
+
+  librbd::group_snap_info2_t gp_snap_info;
+  ASSERT_EQ(-ENOENT, m_rbd.group_snap_get_info(m_ioctx, gp_name, "not_present",
+                                               &gp_snap_info));
+
+  ASSERT_EQ(0, m_rbd.group_snap_get_info(m_ioctx, gp_name, gp_snap_name,
+                                         &gp_snap_info));
+  ASSERT_EQ(gp_snap_name, gp_snap_info.name);
+  ASSERT_EQ(1U, gp_snap_info.image_snaps.size());
+  ASSERT_EQ(image_name, gp_snap_info.image_snaps[0].image_name);
+  ASSERT_EQ(ioctx2.get_id(), gp_snap_info.image_snaps[0].pool_id);
+
+  ASSERT_EQ(0, m_rbd.group_snap_remove(m_ioctx, gp_name, gp_snap_name));
+  ASSERT_EQ(0, m_rbd.group_remove(m_ioctx, gp_name));
+  ASSERT_EQ(0, _rados.pool_delete(pool_name2.c_str()));
+}
+
+TEST_F(TestGroup, snap_list2)
+{
+  REQUIRE_FORMAT_V2();
+
+  std::string pool_name2 = get_temp_pool_name("test-librbd-");
+  ASSERT_EQ(0, rados_pool_create(_cluster, pool_name2.c_str()));
+  BOOST_SCOPE_EXIT(pool_name2) {
+    ASSERT_EQ(0, rados_pool_delete(_cluster, pool_name2.c_str()));
+  } BOOST_SCOPE_EXIT_END;
+
+  rados_ioctx_t ioctx;
+  rados_ioctx_create(_cluster, _pool_name.c_str(), &ioctx);
+  BOOST_SCOPE_EXIT(&ioctx) {
+    rados_ioctx_destroy(ioctx);
+  } BOOST_SCOPE_EXIT_END;
+
+  rados_ioctx_t ioctx2;
+  rados_ioctx_create(_cluster, pool_name2.c_str(), &ioctx2);
+  BOOST_SCOPE_EXIT(&ioctx2) {
+    rados_ioctx_destroy(ioctx2);
+  } BOOST_SCOPE_EXIT_END;
+
+  const char *image_name1="snap_list_image1";
+  uint64_t features;
+  int order =0;
+  ASSERT_TRUE(get_features(&features));
+  ASSERT_EQ(0, rbd_create2(ioctx, image_name1, m_image_size, features,
+                           &order));
+
+  const char *image_name2="snap_list_image2";
+  ASSERT_EQ(0, rbd_create2(ioctx2, image_name2, m_image_size, features,
+                           &order));
+
+  const char *gp_name = "snap_group";
+  ASSERT_EQ(0, rbd_group_create(ioctx, gp_name));
+
+  rbd_group_snap_info2_t *gp_snaps = NULL;
+  size_t num_snaps = 10U;
+  gp_snaps = static_cast<rbd_group_snap_info2_t*>(realloc(
+    gp_snaps, num_snaps * sizeof(rbd_group_snap_info2_t)));
+  BOOST_SCOPE_EXIT(&gp_snaps, &num_snaps) {
+    if (gp_snaps) {
+      rbd_group_snap_list2_cleanup(gp_snaps, num_snaps);
+      free(gp_snaps);
+    }
+  } BOOST_SCOPE_EXIT_END;
+
+  ASSERT_EQ(0, rbd_group_snap_list2(ioctx, gp_name, gp_snaps, &num_snaps));
+  ASSERT_EQ(0U, num_snaps);
+
+  const char *gp_snap_name0 = "snap_snapshot0";
+  ASSERT_EQ(0, rbd_group_snap_create(ioctx, gp_name, gp_snap_name0));
+
+  const char *gp_snap_name1 = "snap_snapshot1";
+  ASSERT_EQ(0, rbd_group_image_add(ioctx, gp_name, ioctx, image_name1));
+  ASSERT_EQ(0, rbd_group_snap_create(ioctx, gp_name, gp_snap_name1));
+
+  const char *gp_snap_name2 = "snap_snapshot2";
+  ASSERT_EQ(0, rbd_group_image_remove(ioctx, gp_name, ioctx, image_name1));
+  ASSERT_EQ(0, rbd_group_image_add(ioctx, gp_name, ioctx2, image_name2));
+  ASSERT_EQ(0, rbd_group_snap_create(ioctx, gp_name, gp_snap_name2));
```

Comment:
For completeness, I'd take one more snapshot here -- with both images in the group.